Add a rootless bootstrap mode using bubblewrap

doraskayo opened the pull request:
This PR adds the ability to bootstrap using the `bwrap` utility, which can operate in rootless mode when user namespaces are supported.

Most of the commits in this PR rework the Python scripts to allow better integration of additional bootstrap modes, and some fix existing issues. The most significant change is that the `prepare()` methods of sysa and sysc are now called directly from `rootfs.py`, and provided with the configuration that each bootstrap mode expects.

See commit messages for more information.

stikonas commented on rootfs.py:
```diff
@@ -139,6 +143,45 @@ def bootstrap(args, system_a, system_b, system_c):
 
+    elif args.bwrap:
+        find_bwrap = """
+import shutil
+print(shutil.which('bwrap'))
+"""
```
Do we need this hack for `bwrap`? Can we not rely on `PATH`?

for `chroot` we couldn't rely on PATH because `chroot` might be in sbin which is not in PATH.

Also as discussed, sort out removing of tmpdir.

Rest of the stuff looks good to me.

fosslinux commented:
Ohter than what stikonas pointed out, lgtm

doraskayo commented:
Thanks for the review.

The `which` hack was removed, and `tmp` directories are now removed recursively. I also added myself under `SPDX-FileCopyrightText`.
